e2e-polybft/framework/test-cluster.go

```diff
@@ -0,0 +1,436 @@
+package framework
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	// envE2ETestsEnabled signal whether the e2e tests will run
+	envE2ETestsEnabled = "E2E_TESTS"
+
+	// envLogsEnabled signal whether the output of the nodes get piped to a log file
+	envLogsEnabled = "E2E_LOGS"
+
+	// envLogLevel specifies log level of each node
+	envLogLevel = "E2E_LOG_LEVEL"
+
+	// envStdoutEnabled signal whether the output of the nodes get piped to stdout
+	envStdoutEnabled = "E2E_STDOUT"
+
+	// accountPassword is the default account password
+	accountPassword = "qwerty"
+)
+
+var startTime int64
+
+func init() {
+	startTime = time.Now().UnixMilli()
+}
+
+func resolveBinary() string {
+	bin := os.Getenv("EDGE_BINARY")
+	if bin != "" {
+		return bin
+	}
+	// fallback
+	return "polygon-edge"
+}
+
+func createAccountPasswordFile(t *testing.T) (string, func()) {
+	t.Helper()
+
+	pwdFile, err := ioutil.TempFile("", "e2e-polybft")
```

Comment:
FYI, you can use `os.CreateTemp` instead of `ioutil.TempFile`. ioutil package is deprecated.

Reply:
🎯